src/plugins/vis_type_timelion/public/components/timelion_vis_component.tsx

```diff
@@ -378,32 +147,87 @@ function TimelionVisComponent({
     [fireEvent]
   );
 
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plotselected').on('plotselected', plotSelectedHandler);
-    }
-  }, [chartElem, plotSelectedHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('mouseleave').on('mouseleave', mouseLeaveHandler);
-    }
-  }, [chartElem, mouseLeaveHandler]);
-
-  useEffect(() => {
-    if (chartElem) {
-      $(chartElem).off('plothover').on('plothover', plotHoverHandler);
-    }
-  }, [chartElem, plotHoverHandler]);
+  const onRenderChange = useCallback(
+    (isRendered: boolean) => {
+      if (!isRendered) {
+        renderComplete();
+      }
+    },
+    [renderComplete]
+  );
 
   const title: string = useMemo(() => last(compact(map(seriesList.list, '_title'))) || '', [
     seriesList.list,
   ]);
 
   return (
-    <div ref={elementRef} className="timChart">
+    <div className="timelionChart">
       <div className="chart-top-title">{title}</div>
-      <div ref={canvasRef} className="chart-canvas" />
+      <Chart renderer="canvas" size={{ width: '100%' }}>
+        <Settings
+          onBrushEnd={brushEndListener}
+          showLegend
+          legendPosition={getLegendPosition()}
+          onRenderChange={onRenderChange}
+          theme={[
+            {
+              crosshair: {
+                band: {
+                  fill: options.crosshair.color,
+                },
+                line: {
+                  strokeWidth: options.crosshair.lineWidth,
+                },
+              },
+            },
+          ]}
+          tooltip={{
+            snap: true,
+            type: TooltipType.VerticalCursor,
+          }}
+        />
+        <Axis
+          id="bottom"
+          position={Position.Bottom}
+          showOverlappingTicks
+          tickFormat={options.xaxis.tickFormatter}
+        />
+        {chart[0]._global && chart[0]._global.yaxes ? (
+          chart[0]._global.yaxes.map((axis: IAxis, index: number) => {
+            return (
+              <Axis
+                key={index}
+                id={axis.position + axis.axisLabel}
+                title={axis.axisLabel}
+                position={axis.position}
+                tickFormat={axis.tickFormatter}
+                gridLine={{
+                  stroke: 'rgba(125,125,125,0.3)',
+                  visible: true,
+                }}
+                domain={axis.domain as YDomainRange}
+              />
+            );
+          })
+        ) : (
+          <Axis
+            id="left"
+            position={Position.Left}
+            gridLine={{
+              stroke: 'rgba(125,125,125,0.3)',
+              visible: true,
+            }}
+          />
```

Comment:
it's too difficult to read. Maybe we should name it somehow or move into separate component?

Reply:
I don't think it's needed. Create a separate document just for rendering Axis it's strange for me. Maybe do you want that I create s separate function for rendering Axis?